### PR TITLE
Adding tabbed list feature for web-receiver Daily Data list

### PR DESCRIPTION
**Pull request by kevinhaube**
This PR adds a tab feature to the Daily Data list for Receivers. It allows Receivers to view data sent by an organization on its own exclusive list rather than all on a single list.

## Changes
- Introduced the `ReportController` class to house functions needed to alter `ReportResource` objects
- Removed `useResource()` hook from `<ReportLink>` in favor of passing the report in via props
- `<TableReports>` was updated to handle getting the list of reports from the API in my last PR (#2108) and now also parses the result and uses `getListOfSenders(report: ReportResource[])` to get an array of unique senders to pass to `<TableButtonGroup>` for labeling
- Added `useState()` hooks to `<TableReports>` and `<TableButtonGroup` with a callback function to sync the two as the buttons are clicked. This controls which sender's reports are displayed in the table.
- Filtering `reports` in `<TableReports>` based on active button (state)
- Only show `<TableButtonGroup>` when `senders` is greater than 1
- UPDATE: Adds `sendingOrg` property to API return values (still need senders to apply property when sending data)
- UPDATE: Handles refactoring and prop typing for `Daily.tsx` into single-component files (thanks @TomNUSDS for suggesting prop typing!)

### Testing
- [X] Tested locally?
- [X] Ran `quickTest all`?
- [X] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [ ] Added tests?

### Specific Security-related subjects a reviewer should pay specific attention to

- Does this PR introduce new endpoints?
No
- Does this PR include changes in authentication and/or authorization of existing endpoints?
No
- Does this change introduce new dependencies that need vetting?
No
- Does this change require changes to our infrastructure?
No
- Does logging contain sensitive data?
No
- Does this PR include or remove any sensitive information itself?
No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Fixes
*List GitHub issues this PR fixes*
- #2036 
- #2128 (partial)

### CODE REVIEW

File: frontend-react/src/pages/Daily.tsx [hunk 1]
```diff
@@ -5,6 +5,9 @@ import OrganizationResource from "../resources/OrganizationResource";
 import { useOktaAuth } from "@okta/okta-react";
 import { groupToOrg } from "../webreceiver-utils";
 import download from "downloadjs";
+import { getListOfSenders } from "../controllers/ReportController";
+import { ButtonGroup, Button } from "@trussworks/react-uswds"
+import { useState } from "react";
 
 const TableData = (props) => {
```
Comment (TomNUSDS) [inline frontend-react/src/pages/Daily.tsx:12]: Not part of this PR, but it would be nice to get a type other than `*any` here for the props at some point. 
(e.g. below, is `props.report.content` required or could report be `null`?)

File: frontend-react/src/pages/Daily.tsx [hunk 2]
```diff
@@ -33,46 +36,55 @@ const TableData = (props) => {
...
             ))}
         </tbody>
     );
 };
 
-const ReportLink = ({ reportId }) => {
-    let report = useResource(ReportResource.detail(), { reportId });
-
-    console.log(report);
+const ReportLink = (props) => {
```
Comment (TomNUSDS) [inline frontend-react/src/pages/Daily.tsx:47]: Again, a nice to have in the future might be to define types for props.

**Review by TomNUSDS — APPROVED** [review]
There's one possible thing to fix and some general feedback, but it LGTM.

I can't really test locally without figuring out how to populate my server with data.

File: frontend-react/src/pages/Daily.tsx (hunk 1 above)
Comment (kevinhaube) [reply frontend-react/src/pages/Daily.tsx:12]: I wouldn't be against refactoring our `src/pages` files a bit; we also have some duplicated components like `<ReportLink>` which exists in `Daily.tsx` and `Details.tsx`. During that process, I'll add Prop types to each component.

**Comment by kevinhaube** [comment]
@TomNUSDS I've refactored `Daily.tsx` and included prop typing for each component. Thank you for the suggestion! Much better for documentation and for any forthcoming front-end people :+1:

File: frontend-react/src/pages/Daily.tsx (hunk 1 above)
Comment (TomNUSDS) [reply frontend-react/src/pages/Daily.tsx]: Sounds like a great plan.
